react/week3/todo-app/src/todolist.js

```diff
@@ -0,0 +1,171 @@
+import { useState, useEffect } from "react";
+import PropTypes from 'prop-types';
+import Timer from "./timer";
+
+const API_URL =
+  "https://gist.githubusercontent.com/benna100/391eee7a119b50bd2c5960ab51622532/raw";
+
+function TodoItem({ id, description, deadline, onDelete, onUpdate }) {
+  const [isChecked, setIsChecked] = useState(false);
+  const [isEdit, setIsEdit] = useState(false);
+  const [updatedDescription, setUpdatedDescription] = useState(description);
+
+  const check = () => {
+    setIsChecked(!isChecked);
+  };
+
+  return (
+    <li className={isChecked ? "checked" : "not-ckecked"}>
+      { isEdit
+      ? <input
+            type="text"
+            value={updatedDescription}
+            onChange={(event)=> {
+                const value = event.target.value;
+                setUpdatedDescription(value);
+            }}/>
+      : <span>#{id} | {description} | {deadline}</span> }
+
+      <input type="checkbox" onChange={check} />
+      <button onClick={() => onDelete(id)}>Delete</button>
+
+      { isEdit
+      ? <button onClick={() => {
+            onUpdate(id, updatedDescription);
+            setIsEdit(false);
+            }
+        }>Update</button>
+      : <button onClick={() => setIsEdit(true)}>Edit</button> }
+
+    </li>
+  );
+}
+
+TodoItem.propTypes = {
+    id: PropTypes.number.isRequired,
+    description: PropTypes.string.isRequired,
+    deadline: PropTypes.string.isRequired,
+    onDelete: PropTypes.func.isRequired,
+    onUpdate: PropTypes.func.isRequired,
+}
+
+function TodoList({ todos, onDelete, onUpdate }) {
+  return (
+    <ul>
+      {todos.length
+        ? todos.map((todo) => (
+            <FancyBorder key={todo.id}>
+              <TodoItem
+                id={todo.id}
+                description={todo.description}
+                deadline={todo.deadline}
+                onDelete={onDelete}
+                onUpdate={onUpdate}
+              ></TodoItem>
+            </FancyBorder>
+          ))
+        : "No items"}
+    </ul>
+  );
+}
+
+function FancyBorder(props) {
```

Comment:
I'd move these additional components into their own files.
Eg. 
`TodoItem.js`
`TodoList.js`
`FancyBorder.js`

This will make this and the other files easier to scroll through. 
And you can reuse the components. For example, the FancyBorder might be used in a different component as well.